src/units.jl

```
@@ -232,19 +232,23 @@ function basefactor(inex, ex, eq, tens, p)
             # Note that sometimes x^1 can cause an overflow error if x is large because
             # of how power_by_squaring is implemented for Rationals, so we use dpow.
             x = dpow(eq*ex*(10//1)^tens, p)
-            return (inex^p, isinteger(x) ? Int(x) : x)
+            result = (inex^p, isinteger(x) ? Int(x) : x)
         else
             x = dpow(ex*(10//1)^tens, p)
-            return ((inex*eq)^p, isinteger(x) ? Int(x) : x)
+            result = ((inex * eq)^p, isinteger(x) ? Int(x) : x)
         end
     else
         if eq_is_exact && can_exact2
-            x = dpow(eq,p)
-            return ((inex * ex * 10.0^tens)^p, isinteger(x) ? Int(x) : x)
+            x = dpow(eq, p)
+            result = ((inex * ex * 10.0^tens)^p, isinteger(x) ? Int(x) : x)
         else
-            return ((inex * ex * 10.0^tens * eq)^p, 1)
+            result = ((inex * ex * 10.0^tens * eq)^p, 1)
         end
     end
+    if isfinite(inex) && !isfinite(first(result)) || !iszero(inex) && iszero(first(result))
```

Why does this only throw an error if `inex` is finite? Wouldn’t it be better to throw an error if `inex` is infinite, too?

---

The idea was to catch the case where the `basefactor()` function itself is the cause of the non-finite value *for valid inputs*, i. e. to detect if the computations in `basefactor()` end up producing something non-finite even if the passed value of `inex` is finite. I don’t know if there could be any valid use case for non-finite `inex`, but if this shouldn’t be allowed, then any checks for this should be part of initial argument validation at the beginning of the function. The error message about floating point overflow/underflow also wouldn’t make sense if `inex` is already non-finite.

---

Ah right, I didn’t see that `inex` was an argument to this method and comes directly from the `basefactors` dict.